File: LV0/배열 만들기 4.js
function solution(arr) {
  let answer = [];
  for (let i = 0; i < arr.length; i++) {
    if (answer.length === 0) {
      answer.push(arr[i]);
    } else if (answer[answer.length - 1] < arr[i]) {
      answer.push(arr[i]);
    } else {
      while (answer.length > 0 && answer[answer.length - 1] >= arr[i]) {
        answer.pop();
      }
      answer.push(arr[i]);
    }
  }
  return answer;
}

File: LV0/배열 만들기 4.java
class Solution {

  public int[] solution(int[] arr) {
    List<Integer> answer = new ArrayList<>();
    for (int i = 0; i < arr.length; i++) {
      if (answer.size() == 0) {
        answer.add(arr[i]);
      } else if (answer.get(answer.size() - 1) < arr[i]) {
        answer.add(arr[i]);
      } else {
        while (answer.size() > 0 && answer.get(answer.size() - 1) >= arr[i]) {
          answer.remove(answer.size() - 1);
        }
        answer.add(arr[i]);
      }
    }
    return answer.stream().mapToInt(i -> i).toArray();
  }
}
